CharlieBackend.Api.UnitTes/ScheduleServiceTests.cs

```diff
@@ -0,0 +1,649 @@
+                Pattern = new PatternForCreateScheduleDTO
+                {
+                    Type = PatternType.Daily,
+                    Interval = 1,
+                    DaysOfWeek = new List<DayOfWeek> { DayOfWeek.Monday },
+                },
+
+                Range = new OccurenceRange
+                {
+                    StartDate = DateTime.Parse("Jan 1, 2021"),
+                    FinishDate = DateTime.Parse("Feb 1, 2021")
+                },
+
+                Context = new ContextForCreateScheduleDTO
+                {
+                    GroupID = 3,
+                    MentorID = 1,
+                    ThemeID = 5
+                }
+            };
+
+            validEventOccurrence = new EventOccurrence
+            {
+                Pattern = validScheduleDTO.Pattern.Type,
+                StudentGroupId = validScheduleDTO.Context.GroupID,
+                EventStart = validScheduleDTO.Range.StartDate,
+                EventFinish = validScheduleDTO.Range.FinishDate.Value,
+                Storage = EventOccuranceStorageParser.GetPatternStorageValue(validScheduleDTO.Pattern)
+            };
+        }
+
+        private void Initialize(CreateScheduleDto createScheduleDto, Mock<IThemeRepository> themeRepositoryMock = null,
+            Mock<IMentorRepository> mentorRepositoryMock = null, Mock<IStudentGroupRepository> studentGroupRepositoryMock = null)
+        {
+            _scheduleRepositoryMock.Setup(x => x.AddRange(new List<ScheduledEvent>()
+            {
+                new ScheduledEvent
+                {
+                    ThemeId = 5,
+                    StudentGroupId = 3,
+                    MentorId = 1
+                }
+            }
+            ));
+
+            _eventOccuranceRepositoryMock.Setup(x => x.Add(
+                new EventOccurrence
+                {
+                    StudentGroupId = 3,
+                    EventStart = createScheduleDto.Range.StartDate,
+                    Pattern = createScheduleDto.Pattern.Type
+                }
+                ));
+
+            if (themeRepositoryMock != null)
+            {
+                _unitOfWorkMock.Setup(x => x.ThemeRepository).Returns(themeRepositoryMock.Object);
+            }
+
+            if (mentorRepositoryMock != null)
+            {
+                _unitOfWorkMock.Setup(x => x.MentorRepository).Returns(mentorRepositoryMock.Object);
+            }
+
+            if (studentGroupRepositoryMock != null)
+            {
+                _unitOfWorkMock.Setup(x => x.StudentGroupRepository).Returns(studentGroupRepositoryMock.Object);
+            }
+        }
+
+        [Fact]
+        public async Task CreateScheduleAsync_ValidScheduleRequest_ShouldReturnEventOccurrenceDTO()
+        {
+            var themeRepositoryMock = new Mock<IThemeRepository>();
+            themeRepositoryMock.Setup(x => x.IsEntityExistAsync(5)).ReturnsAsync(true);
+
+            var mentorRepositoryMock = new Mock<IMentorRepository>();
```

Comment:
Move this initialization to the constructor.

Reply:
Do the same for other mocks.

Reply:
Ok, done

Reply:
https://github.com/ita-social-projects/WhatBackend/pull/422/commits/4d0738f7f779ccd964c808782d973bc3255869ee